serve/handlers/subs/subs.go

```diff
@@ -164,27 +183,45 @@ func (h *Handler) GetFilterChangesHandler(_ *metadata.Metadata, params []any) (a
 		return nil, spec.GenerateResponseError(err)
 	}
 
-	// Handle block filter changes
-	changes := h.getBlockChanges(f)
+	// Handle filter changes
+	changes, err := h.getFilterChanges(f)
+	if err != nil {
+		return nil, spec.GenerateResponseError(err)
+	}
 
-	// Encode the response
-	encodedResponses := make([]string, len(changes))
+	results := make([]string, len(changes))
 
-	for index, change := range changes {
-		encodedResponse, encodeErr := encode.PrepareValue(change)
+	for index, changed := range changes {
+		encodedResponse, encodeErr := encode.PrepareValue(changed)
 		if encodeErr != nil {
 			return nil, spec.GenerateResponseError(encodeErr)
 		}
 
-		encodedResponses[index] = encodedResponse
+		results[index] = encodedResponse
 	}
 
-	return encodedResponses, nil
+	return results, nil
 }
 
-func (h *Handler) getBlockChanges(filter filters.Filter) []types.Header {
+func (h *Handler) getFilterChanges(filter filters.Filter) ([]any, error) {
 	// Get updates
-	blockHeaders, _ := filter.GetChanges().([]types.Header)
+	changes := filter.GetChanges()
+	value := reflect.ValueOf(changes)
```

Comment:
This reflect segment can be completely dropped -- the filter gives you the `any` type already, you just need to return it

Reply:
Thank you for your suggestion for a simple fix.